### PR TITLE
User posts

### LINKED ISSUE #10: Allow users to create posts
Posts should have:

- Text
- created_at/updated_at/soft_deletion
- Foreign key to Users

### PR DESCRIPTION
**Pull request by EdiOancea**
Closes #10

### CODE REVIEW

File: app/models/post.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+from app.models.abstracts import Date
+from app.models import User
+from app.managers import UserManager
+
+class Post(Date):
+    class Meta:
+        db_table = 'posts'
+
+    REQUIRED_FIELDS = ['content']
```
Comment (raresluncan) [inline app/models/post.py]: I think this can be removed since not specifying required=True or required=False when initialising a field renderds the field required. For example

user = Models.ForeignKey(User, .... ) makes the user fk required but

user = Models.foreignKey(User ,... , required=false) creates a nullable fk in the database.

File: app/tests/test_auth.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import json
+
+from django.test import TestCase
+from rest_framework.test import APIClient
+
+class UserTestCase(TestCase):
```
Comment (raresluncan) [inline app/tests/test_auth.py]: We can also test with empty/ incomplete payload. (Ex: login with 'email' but provide not password attribute or vice versa).

File: app/tests/test_user.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+from django.test import TestCase
+from rest_framework.test import APIClient
+
+from app.models import User
+from app.serializers import UserSerializer
+
+class UserTestCase(TestCase):
+    fixtures = ('user',)
+    SERVER_NAME = 'http://gucci.com/api'
```
Comment (raresluncan) [inline app/tests/test_user.py]: 400593_AP00T_1000_001_100_0000_Light-Leather-belt-with-Double-G-buckle

Let's use an appropriate name (we will kill this easter egg just for debugging purposes).

File: app/views/post.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+from rest_framework import viewsets, permissions
+from rest_framework.response import Response
+from rest_framework.decorators import action
+
+from app.models import Post
+from app.serializers import PostSerializer
+
+class PostViewSet(viewsets.ModelViewSet):
+    serializer_class = PostSerializer
+    permission_classes = (permissions.AllowAny,)
```
Comment (raresluncan) [inline app/views/post.py]: Let's start adding permissions here. I presume the user must be at least logged in to post something.

**Review by raresluncan — COMMENTED** [review]
We should also add tests for Post entity.

Look into setUp method and SERVER_NAME attributes. Could we create a common class so all tests inherit SERVER_NAME attribute and a set_up with a regular user client?

File: app/tests/test_user.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+    def setUp(self):
+        self.client = APIClient()
+        self.user = User.objects.get(email='eduard.oancea@algotech.solutions')
+        self.client.force_authenticate(user=self.user)
+
+    def test_get_one(self):
+        config = { 'url': '/users/1/', 'pk': 1}
```
Comment (raresluncan) [inline app/tests/test_user.py:14]: What happens when there is an invalid id in url (for get_one, or update, or destroy a.k.a delete)?

Cases:
- invalid id is number (Ex: /users/243424/)
- invalid id is string (Ex: /users/invalid_id/)]

We should also add tests for those.

File: app/tests/test_auth.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    fixtures = ('user',)
+    SERVER_NAME = 'http://gucci.com/api'
+
+    def setUp(self):
```
Comment (raresluncan) [inline app/tests/test_auth.py]: set_up (no camel case in python methods)
